edb/server/dbview/dbview.pyx

```diff
@@ -555,7 +555,7 @@ cdef class DatabaseIndex:
         self._comp_sys_config = config.get_compilation_config(sys_config)
 
     def get_db(self, dbname):
-        return self._dbs[dbname]
+        return self._dbs.get(dbname)
```

Comment:
I used `[]` on purpose here, to let it fail (it never should fail, unless there's a bug).

Reply:
`POST /db/nonexistent/edgeql` crashes with a `KeyError` currently, and I don't think that catching `KeyError` is nicer than checking for `None`.  We can add another method: `maybe_get_db` or something.

Reply:
Well, you might as well then fix connecting to a non-existent DB via binary.

Reply:
Returning None here might make it a segfault now. Please check all get_db() callers.

Reply:
Also, I wouldn't return None. I'd raise a proper `edb.errors` exception here, or at least a LookupError.

Reply:
Done.